### ISSUE TITLE
Consider treating \r in shell as TTY-s do

### ISSUE THREAD

**Issue by aivarannamaa** [thonny/thonny#269]
**[Original report](https://bitbucket.org/bitbucket-issue-migration\thonny-issues.zip/issue/269) by me.**

----------------------------------------

Check following program in console:

```
from time import sleep
import sys

for i in range (101):
    sys.stdout.write("Download progress: %d%%   \r" % (i) )
    sys.stdout.flush()
    sleep(0.1)
```

**Comment by aivarannamaa** [comment]
I wouldn't consider it a bug. Thonny's shell is not a TTY, neither is IDLE's shell, for example. Also, if you run `python sample.py > out.txt` you would get similar result as Thonny produces.

I agree it would be a nice thing to have, but most of the terminal stuff would be quite cumbersome to implement, especially because the nice debugger allows taking back steps and the shell needs to take this into account.

Did you know that in Thonny 3.1 you can run current script in system terminal (Run => Run current script in terminal)?

**Comment by phr0gger** [comment]
Understood. Run in terminal is a solution. I was still on 3.08. Time to update the link on Thonny.org!

**Comment by aivarannamaa** [comment]
Fixed in 3.2
